src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HearingsUpdateHearingRequest.java

```diff
@@ -0,0 +1,105 @@
+
+    private final String hearingsApiEndpoint;
+    private final String aboutToStartPath;
+    private final String midEventPath;
+
+    AsylumCaseCallbackApiDelegator asylumCaseCallbackApiDelegator;
+
+    public HearingsUpdateHearingRequest(
+            AsylumCaseCallbackApiDelegator asylumCaseCallbackApiDelegator,
+            @Value("${hearingsApi.endpoint}") String hearingsApiEndpoint,
+            @Value("${hearingsApi.aboutToStartPath}") String aboutToStartPath,
+            @Value("${hearingsApi.midEventPath}") String midEventPath
+    ) {
+        this.asylumCaseCallbackApiDelegator = asylumCaseCallbackApiDelegator;
+        this.hearingsApiEndpoint = hearingsApiEndpoint;
+        this.aboutToStartPath = aboutToStartPath;
+        this.midEventPath = midEventPath;
+    }
+
+    @Override
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_START
+                || callbackStage == PreSubmitCallbackStage.MID_EVENT)
+                && Objects.equals(Event.UPDATE_HEARING_REQUEST, callback.getEvent());
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase;
+
+        if (callback.getCaseDetails().getCaseData().read(UPDATE_HEARINGS).isEmpty()) {
+            asylumCase = getHearings(callback);
+        } else {
+            asylumCase = getHearingDetails(callback);
+            setHearingLocationDetails(asylumCase);
+        }
+
+        return new PreSubmitCallbackResponse<>(asylumCase);
+    }
+
+    private AsylumCase getHearings(Callback<AsylumCase> callback) {
+        return asylumCaseCallbackApiDelegator.delegate(callback, hearingsApiEndpoint + aboutToStartPath);
+    }
+
+    private AsylumCase getHearingDetails(Callback<AsylumCase> callback) {
+        return asylumCaseCallbackApiDelegator.delegate(callback, hearingsApiEndpoint + midEventPath);
+    }
+
+    private static void setHearingLocationDetails(AsylumCase asylumCase) {
+        Optional<String> hearingLocation = asylumCase.read(HEARING_LOCATION_VALUE);
+        if (hearingLocation.isPresent()) {
+            String hearingCenterValue = HearingCentre.getValueByEpimsId(hearingLocation.get());
+            asylumCase.write(HEARING_LOCATION_VALUE, hearingCenterValue);
+            DynamicList locationsList = new DynamicList(
+                    new uk.gov.hmcts.reform.iacaseapi.domain.entities.Value(hearingLocation.get(), hearingCenterValue),
```

Comment:
Can we import the class?

Reply:
I can't it collides with the Value spring annotation https://github.com/hmcts/ia-case-api/blob/e44cbd36298166463ddfe3312f1ec2e9eef3657b/src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HearingsUpdateHearingRequest.java#L37-L37